### PR TITLE
Mes 1955 view cat b manoeuvres overlay

### PR DESCRIPTION
**Pull request by Jason-Pullen**
## Description and relevant Jira numbers
Mes 1955 add a new competency type which has a popover for content to be added later.

added new manoeuvres control which has a popover.
added holder for manoeuvre-popover component.
added tick indicator component
changed duplicate ids to classes
fixed layout issues on competences
## Pull Request checklist:

- [x] [WIP] tag removed from PR title
- [x] PR has an understandable description

## Git feature branch checklist:

- [x] branch name comply with our branching strategy
- [x] git branch contains relevant JIRA ticket number
- [ ] branch rebased against the latest develop
- [ ] commits are squashed

## Sign off process checklist:

- [x] Code has been tested manually
- [x] Tests are passing
- [x] PR link added to JIRA
- [ ] Reviewers selected in Github
- [ ] Any new reusable component/widget added to component library on Confluence

- [x] Screenshot(s) captured on the physical device (if applicable)
- [x] Tested by QA
- [ ] PO's approval
Screenshot 2019-03-28 at 14 45 45
Screenshot 2019-03-28 at 14 45 55

### CODE REVIEW

File: src/modules/tests/test_data/test-data.constants.ts [hunk 1]
```diff
@@ -1,4 +1,5 @@
 export enum Competencies {
+  manoeuvres = 'manoeuvres',
```
Comment (Cavalier07) [inline src/modules/tests/test_data/test-data.constants.ts]: Not sure that manoeuvres should be put here as this maps directly to the competencies in the schema , manoeuvres lives in a different part of the schema

File: src/pages/test-report/__tests__/test-report.spec.ts [hunk 2]
```diff
@@ -27,7 +28,9 @@ describe('TestReportPage', () => {
 
   beforeEach(async(() => {
     TestBed.configureTestingModule({
-      declarations: [
+      declarations: [TestReportPage,
+        MockComponent(CompetencyComponent),
```
Comment (Cavalier07) [inline src/pages/test-report/__tests__/test-report.spec.ts]: This one is already in here (2 lines below) and ManoeuvresPopoverComponent should be MockComponent(ManoeuvresPopoverComponent)

File: src/pages/test-report/components/competency/competency.ts [hunk 3]
```diff
@@ -42,14 +47,27 @@ export class CompetencyComponent {
   competencyState: ComptencyState;
   subscription: Subscription;
   faultCount: number;
-
+  promoted: boolean;
```
Comment (johnmorleybjss) [inline src/pages/test-report/components/competency/competency.ts]: I don't really follow what `promoted` means in this context, perhaps we can come up with a better name?

File: src/pages/test-report/test-report.scss [hunk 4]
```diff
@@ -1,29 +1,81 @@
 page-test-report {
   .content-ios {
-    background-color: map-get($colors-gds, 'gds-grey-3');
+    background-color: map-get($colors-gds, "gds-grey-3");
```
Comment (szabolcs-keresztes) [inline src/pages/test-report/test-report.scss]: Single quote was good

File: src/pages/test-report/components/competency/competency.html [hunk 5]
```diff
@@ -1,9 +1,27 @@
-<button #competencyButton>
-  <fault-counter id="drivingFaults" [count]="faultCount"></fault-counter>
-  <span id='competencyLabel'>{{ getLabel() }}</span>
-  <div>
+<button #competencyButton (click)="togglePromote()" [class.promoted]="promoted">
```
Comment (Cavalier07) [inline src/pages/test-report/components/competency/competency.html]: I feel that we should have 2 buttons, the competency button  and a specific one for manoeuvres. It seems like this one button is going to be getting complicated with fault counts, tick indicators, popups, different styled labels and all the logic around them all

**Pull request merged**